server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeService.java

```diff
@@ -0,0 +1,154 @@
+    public static final Setting<CompatibilityMode> REMOTE_STORE_COMPATIBILITY_MODE_SETTING = new Setting<>(
+        "remote_store.compatibility_mode",
+        CompatibilityMode.STRICT.name(),
+        CompatibilityMode::parseString,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    /**
+     * Node join compatibility mode introduced with remote backed storage.
+     *
+     * @opensearch.internal
+     */
+    public enum CompatibilityMode {
+        STRICT("strict"),
+        ALLOW_MIX("allow_mix");
+
+        public final String mode;
+
+        CompatibilityMode(String mode) {
+            this.mode = mode;
+        }
+
+        public static CompatibilityMode parseString(String compatibilityMode) {
+            try {
+                return CompatibilityMode.valueOf(compatibilityMode.toUpperCase(Locale.ROOT));
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException(
+                    "["
+                        + compatibilityMode
+                        + "] compatibility mode is not supported. "
+                        + "supported modes are ["
+                        + CompatibilityMode.values().toString()
+                        + "]"
+                );
+            }
+        }
+    }
+
+    public RemoteStoreNodeService(Supplier<RepositoriesService> repositoriesService, ThreadPool threadPool) {
+        this.repositoriesService = repositoriesService;
+        this.threadPool = threadPool;
+    }
+
+    /**
+     * Creates a repository during a node startup and performs verification by invoking verify method against
+     * mentioned repository. This verification will happen on a local node to validate if the node is able to connect
+     * to the repository with appropriate permissions.
+     */
+    public List<Repository> createAndVerifyRepositories(DiscoveryNode localNode) {
+        RemoteStoreNodeAttribute node = new RemoteStoreNodeAttribute(localNode);
+        List<Repository> repositories = new ArrayList<>();
+        for (RepositoryMetadata repositoryMetadata : node.getRepositoriesMetadata().repositories()) {
+            String repositoryName = repositoryMetadata.name();
+
+            // Create Repository
+            RepositoriesService.validate(repositoryName);
+            Repository repository = repositoriesService.get().createRepository(repositoryMetadata);
+            logger.info(
+                "remote backed storage repository with name {} and type {} created.",
+                repository.getMetadata().name(),
+                repository.getMetadata().type()
+            );
+
+            // Verify Repository
+            String verificationToken = repository.startVerification();
+            repository.verify(verificationToken, localNode);
+            repository.endVerification(verificationToken);
+            logger.info(() -> new ParameterizedMessage("successfully verified [{}] repository", repositoryName));
+
```

Comment:
What happens if we create a repository and the verification fails? Will we have a lingering repo?